pep-0999.rst

```diff
@@ -0,0 +1,541 @@
+       else:
+           reveal_type(packet.data)  # Missing? error?
+           packet_bytes = b''
+       socket.send(packet_bytes)
+
+or a check against ``locals()`` for local variables:
+
+::
+
+   def send_data(packet_data: Optional[str]) -> None:
+       packet_bytes: Union[str, Missing]
+       if packet_data is not None:
+           packet_bytes = packet.data.encode('utf-8')
+       
+       if 'packet_bytes' in locals():
+           reveal_type(packet_bytes)  # bytes
+           socket.send(packet_bytes)
+       else:
+           reveal_type(packet_bytes)  # Missing? error?
+
+or a check via other means, such as against ``globals()`` for global
+variables:
+
+::
+
+   warning: Union[str, Missing]
+   import sys
+   if sys.version_info < (3, 6):
+       warning = 'Your version of Python is unsupported!'
+
+   if 'warning' in globals():
+       reveal_type(warning)  # str
+       print(warning)
+   else:
+       reveal_type(warning)  # Missing? error?
+
+Weird and inconsistent. ``Missing`` is not really a value at all; it’s
+an absence of definition and such an absence should be treated
+specially.
+
+Difficult to implement
+''''''''''''''''''''''
+
+Eric Traut from the Pyright type checker team has stated that
+implementing a ``Union[..., Missing]``-style syntax would be
+difficult. [2]_
+
+Introduces a second null-like value into Python
+'''''''''''''''''''''''''''''''''''''''''''''''
+
+Defining a new ``Missing`` type-level constant would be very close to
+introducing a new ``Missing`` value-level constant at runtime, creating
+a second null-like runtime value in addition to ``None``. Having two
+different null-like constants in Python (``None`` and ``Missing``) would
+be confusing. Many newcomers to JavaScript already have difficulty
+distinguishing between its analogous constants ``null`` and
+``undefined``.
+
+
+Replace Optional with Nullable. Repurpose Optional to mean “optional item”.
+---------------------------------------------------------------------------
+
+``Optional[...]`` is too ubiquitous to deprecate. Although use of it
+*may* fade over time in favor of the ``T|None`` syntax specified by `PEP
+604 <https://www.python.org/dev/peps/pep-0604/>`__.
+
+
+Change Optional to mean “optional item” in certain contexts instead of “nullable”
+---------------------------------------------------------------------------------
+
+Consider the use of a special flag on a TypedDict definition to alter
+the interpretation of ``Optional`` inside the TypedDict to mean
+“optional item” rather than its usual meaning of “nullable”:
+
+::
+
+   class MyThing(TypedDict, optional_as_missing=True):
+       req1: int
+       opt1: Optional[str]
+
+or:
+
+::
+
+   class MyThing(TypedDict, optional_as_nullable=False):
+       req1: int
+       opt1: Optional[str]
+
+This would add more confusion for users because it would mean that in
+*some* contexts the meaning of ``Optional[...]`` is different than in
+other contexts.
```

Comment:
Also, the flag on the type object is easily overlooked.

Reply:
Did add trailing sentence fragment:

> This would add more confusion for users because it would mean that in
*some* contexts the meaning of ``Optional[...]`` is different than in
other contexts, **and it would be easy to overlook the flag.**